include/swift/Driver/DependencyGraph.h

```diff
@@ -264,6 +273,15 @@ class DependencyGraph : public DependencyGraphImpl {
   ///
   /// If you want to see how each node gets added to \p visited, pass a local
   /// MarkTracer instance to \p tracer.
+  ///
+  /// Assumes that the starting node is "cascading"; that all dependencies must
+  /// be dirty if the start is dirty. Therefore, mark the start. For each
+  /// visited node, add it to visited, and mark it if it cascades. The start
+  /// node is NOT added to visited.
+  ///
+  /// Do not confused "Marked" with "Visited".
+  /// "Marked" does NOT influence the traversal. The traversal routines use
+  /// "Visited" to avoid endless "recursion".
```

Comment:
This isn't correct; already-marked nodes do not end up in `visited`, and it's not just to avoid endless recursion. I got that wrong.

(Also, another Doxygen note: parameters can be referred to inline with `\p visited`.)

Reply:
Good catch! Fixed.